test/org/zaproxy/zap/extension/pscanrules/XFrameOptionsScannerUnitTest.java

```diff
@@ -0,0 +1,333 @@
+		
+		msg.setResponseBody("<html></html>");
+		msg.setResponseHeader(
+				"HTTP/1.1 401 OK\r\n" +
+				"Server: Apache-Coyote/1.1\r\n" +
+				"Content-Type: text/html;charset=ISO-8859-1\r\n" +
+				"Content-Length: " + msg.getResponseBody().length() + "\r\n");
+		String response = msg.getResponseHeader().toString() + msg.getResponseBody().toString();
+		Source source = new Source(response);
+		rule.scanHttpResponseReceive(msg, rule.getPluginId(), source);
+
+		assertThat(alertsRaised.size(), equalTo(0));
+    }
+
+    @Test
+	public void multipleXframeOptions () throws HttpMalformedHeaderException {
+		
+		HttpMessage msg = new HttpMessage();
+		msg.setRequestHeader("GET http://www.example.com/test/ HTTP/1.1");
+		
+		msg.setResponseBody("<html></html>");
+		msg.setResponseHeader(
+				"HTTP/1.1 200 OK\r\n" +
+				"Server: Apache-Coyote/1.1\r\n" +
+				"X-Frame-Options: DENY\r\n" +
+				"X-Frame-Options: SAMEORIGIN\r\n" +
+				"Content-Type: text/html;charset=ISO-8859-1\r\n" +
+				"Content-Length: " + msg.getResponseBody().length() + "\r\n");
+		String response = msg.getResponseHeader().toString() + msg.getResponseBody().toString();
+		Source source = new Source(response);
+		rule.scanHttpResponseReceive(msg, rule.getPluginId(), source);
+
+		assertThat(alertsRaised.size(), equalTo(1));
+		assertThat(alertsRaised.get(0).getParam(), equalTo(""));
+		assertThat(alertsRaised.get(0).getEvidence(), equalTo(""));
+    }
+    
+    @Test
+	public void xframeOptionsViaMetaTag () throws HttpMalformedHeaderException {
+		
+		HttpMessage msg = new HttpMessage();
+		msg.setRequestHeader("GET http://www.example.com/test/ HTTP/1.1");
+		
+		msg.setResponseBody(
+				"<html>" +
+				"<meta http-equiv=\"X-Frame-Options\" content=\"DENY\">"+
+				"</html>");
+		msg.setResponseHeader(
+				"HTTP/1.1 200 OK\r\n" +
+				"Server: Apache-Coyote/1.1\r\n" +
+				"X-Frame-Options: DENY\r\n" +
+				"Content-Type: text/html;charset=ISO-8859-1\r\n" +
+				"Content-Length: " + msg.getResponseBody().length() + "\r\n");
+		String response = msg.getResponseHeader().toString() + msg.getResponseBody().toString();
+		Source source = new Source(response);
+		rule.scanHttpResponseReceive(msg, rule.getPluginId(), source);
+
+		assertThat(alertsRaised.size(), equalTo(1));
+		assertThat(alertsRaised.get(0).getParam(), equalTo(""));
+		assertThat(alertsRaised.get(0).getEvidence(), equalTo(
+				"<meta http-equiv=\"X-Frame-Options\" content=\"DENY\">"));
+    }
+    
+    @Test
+	public void xframeOptionsViaMetaTagAndHeader () throws HttpMalformedHeaderException {
+		
+		HttpMessage msg = new HttpMessage();
+		msg.setRequestHeader("GET http://www.example.com/test/ HTTP/1.1");
+		
+		msg.setResponseBody(
+				"<html>" +
+				"<meta http-equiv=\"X-Frame-Options\" content=\"DENY\">"+
+				"</html>");
+		msg.setResponseHeader(
+				"HTTP/1.1 200 OK\r\n" +
+				"Server: Apache-Coyote/1.1\r\n" +
+				"X-Frame-Options: DENY\r\n" +
+				"X-Frame-Options: SAMEORIGIN\r\n" +
```

Comment:
One of the headers should be removed (the case of multiple headers is already tested).